From 100b3b7972c2a0cf341aa97971f5d7c9d1a0b68b Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Tue, 22 Nov 2016 09:07:54 +0100
Subject: [PATCH] avoid loss of text in edited placeholder

How to reproduce:
1. Create an empty presentation.
2. Save it.
3. Click at one of the text placeholders and write something.
4. Save again.
5. Reload. The placeholder is missing. (Actually, it's still there, but
   empty and 0-size.)

This only happens if a11y is enabled.

Change-Id: I7ae3d3c6ab0ce91aef01c0e9d62f7963dce965fc
---
 svx/source/unodraw/unoshtxt.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/svx/source/unodraw/unoshtxt.cxx b/svx/source/unodraw/unoshtxt.cxx
index cda12f7..3a5705f 100644
--- a/svx/source/unodraw/unoshtxt.cxx
+++ b/svx/source/unodraw/unoshtxt.cxx
@@ -589,6 +589,7 @@ SvxTextForwarder* SvxTextEditSourceImpl::GetBackgroundTextForwarder()
         bCreated = true;
 
         mbForwarderIsEditMode = false;
+        mbDataValid = false;
     }
 
     if( mpObject && mpText && !mbDataValid && mpObject->IsInserted() && mpObject->GetPage() )
-- 
2.9.3

